implementations/rust/ockam/ockam_command/src/project/create.rs

```diff
@@ -50,33 +50,14 @@ async fn run_impl(
     opts: CommandGlobalOpts,
     cmd: CreateCommand,
 ) -> miette::Result<()> {
-    let space_id = opts.state.spaces.get(&cmd.space_name)?.config().id.clone();
     let node = InMemoryNode::start(ctx, &opts.state).await?;
-    let controller = node.create_controller().await?;
-
-    let project = controller
-        .create_project(ctx, space_id, cmd.project_name, vec![])
+    let project = node
+        .create_project(ctx, &cmd.space_name, &cmd.project_name, vec![])
         .await?;
     let operation_id = project.operation_id.clone().unwrap();
+    let controller = node.create_controller().await?;
     check_for_completion(&opts, ctx, &controller, &operation_id).await?;
     let project = check_project_readiness(&opts, ctx, &node, project).await?;
-    opts.state
-        .projects
-        .overwrite(&project.name, project.clone())?;
-    opts.state
-        .trust_contexts
-        .overwrite(&project.name, project.clone().try_into()?)?;
```

Comment:
_Appreciation_

Love how the state management is now hidden behind the controller call!